clcache.py

```diff
@@ -141,6 +144,18 @@ def getManifest(self, manifestHash):
             return None
 
 
+@contextlib.contextmanager
+def allSectionsLocked(repository):
+    sections = list(repository.sections())
```

Comment:
There may be more (or less!) sections after this generator `yield`s, in which case we may end up releasing more or less mutexes than we acquired. I create a copy to make sure that this is in sync.